pilot/pkg/networking/plugin/authn/authentication.go

```diff
@@ -92,13 +96,62 @@ func buildFilter(in *plugin.InputParams, mutable *networking.MutableObjects) err
 
 // OnInboundPassthrough is called whenever a new passthrough filter chain is added to the LDS output.
 func (Plugin) OnInboundPassthrough(in *plugin.InputParams, mutable *networking.MutableObjects) error {
-	return nil
+	if in.Node.Type != model.SidecarProxy {
+		// Only care about sidecar.
+		return nil
+	}
+
+	return buildFilter(in, mutable, true)
 }
 
 // OnInboundPassthroughFilterChains is called for plugin to update the pass through filter chain.
 func (Plugin) OnInboundPassthroughFilterChains(in *plugin.InputParams) []networking.FilterChain {
-	// Pass nil for ServiceInstance so that we never consider any alpha policy for the pass through filter chain.
 	applier := factory.NewPolicyApplier(in.Push, in.Node.Metadata.Namespace, labels.Collection{in.Node.Metadata.Labels})
-	// Pass 0 for endpointPort so that it never matches any port-level policy.
-	return applier.InboundFilterChain(0, constants.DefaultSdsUdsPath, in.Node, in.ListenerProtocol, trustDomainsForValidation(in.Push.Mesh))
+	trustDomains := trustDomainsForValidation(in.Push.Mesh)
+	// First generate the default passthrough filter chains, pass 0 for endpointPort so that it never matches any port-level policy.
+	filterChains := applier.InboundFilterChain(0, constants.DefaultSdsUdsPath, in.Node, in.ListenerProtocol, trustDomains)
+
+	// Then generate the per-port passthrough filter chains.
+	for port := range applier.PortLevelSetting() {
+		// Skip the per-port passthrough filterchain if the port is already handled by OnInboundFilterChains().
+		if portAlreadyHandled(port, in.Node) {
+			continue
+		}
+
+		log.Debugf("Generated extra passthrough filter chain for port: %d", port)
```

Comment:
Should we also log name of service or other unique id from `in` for debug? Current we only have port.

Reply:
Added node.id in the log.